lib/index.js

```diff
@@ -19,23 +31,47 @@ module.exports = (referenceDir, config, { watch, fingerprint, sourcemaps, compac
 		fingerprint,
 		exitOnError: !watch
 	});
-	let watcher = watch && makeWatcher(config.watchDirs, referenceDir,
-			assetManager.resolvePath);
 	let browsers = browserslist.findConfig(referenceDir) || {};
 
 	let plugins = Object.assign({}, DEFAULTS.plugins, config.plugins);
+	let buckets = {
+		static: [],
+		scripts: [],
+		styles: [],
+		markup: []
+	};
 	Object.keys(plugins).forEach(type => {
 		let pluginConfig = config[type];
 		if(!pluginConfig) {
 			return;
 		}
 
-		let plugin = plugins[type];
+		let bucket = plugins[type].bucket;
+		let plugin = plugins[type].package;
 		if(!plugin.call) {
 			plugin = load(plugin);
 		}
-		plugin(pluginConfig, assetManager, { watcher, browsers, sourcemaps, compact });
+		let initializedPlugin = plugin(pluginConfig, assetManager,
+				{ browsers, sourcemaps, compact });
+		buckets[bucket].push(initializedPlugin);
 	});
+
+	let buildStep1 = buckets.static;
```

Comment:
This reassignment is kind of weird. I'm thinking of pushing to the buildStep variables directly.

Reply:
(Assuming by reassignment you mean creating those temporary variables.) I agree that those variable names are awkward; see below.